graphql/models_gen.go

```diff
@@ -84,19 +96,15 @@ type HostsResponse struct {
 	Hosts              []*model.APIHost `json:"hosts"`
 }
 
-type PatchBuildVariant struct {
-	Variant     string           `json:"variant"`
-	DisplayName string           `json:"displayName"`
-	Tasks       []*model.APITask `json:"tasks"`
+type MainlineCommit struct {
+	Version     *model.APIVersion `json:"version"`
+	VersionMeta *VersionMeta      `json:"versionMeta"`
```

Comment:
Yes this is so the buildVariant resolver isn't used on non activated versions. Since the types are different we're able to be explicit about what fields activated and non activated versions accept. I may try to build out a better type for this though.

Reply:
I tried to implement something like this.
```
type MainlineCommit {
  version: Version
}

union Version = UnactivatedVersion | ActivatedVersion

type UnactivatedVersion {
  id: String!
  createTime: Time!
  message: String!
  revision: String!
  author: String!
  status: String!
  order: Int!
  repo: String!
  project: String!
  branch: String!
  requester: String!
  activated: Boolean
}

type ActivatedVersion {
  startTime: Time!
  finishTime: Time!
  buildVariants(options: BuildVariantOptions): [GroupedBuildVariant]
}

```
But a [bug](https://github.com/99designs/gqlgen/issues/1424) in gqlgen does not allow me to generate the types for union types and interfaces. So i think this is the best we can do for now.